code-challenges.test.js

```diff
@@ -9,51 +12,207 @@
+// Psuedocode: create a function named Nounfinder1 and Nounfinder2
+// parameter will be an array named array
+// Input : we want to input an array of strings
+//  Output: we expect to return an array of strings with all of the words capitalized within. 
+// create a function that takes in an array of words and returns array with all the words capitalized
+const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+// // declare a function that takes in an array
+const Nounfinder1 = (array) => {
+    //  return the array output from the map action
+    return array.map ( (value) => {
+        // with each value access the first index and capitalize the letter then concatenate the remainder of the word
+        return value[0].toUppercase() + value.conCat(1)
+    } )
+}
+// I keep getting the     ReferenceError: nounfinder1 is not defined, I've tried a variety of different syntax's to tackle this problem and I believe I'm over thinking the process. I understand the overall concept of testing and creating higher order functions, however I'm struggling with implementation.  I would like for someone to go over the logic behind this process a bit more, perhaps in this weeks office hours.  I've ran the test about 10 different times over the last 3 hours, so at this point I will opt to move on from this problem and tackle another . 
+
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
-
-const vowelTester1 = "learn"
+ describe("firstVowels", () => {
+    it("logs the index of the first vowel", ()=> {
+        const vowelTester1 = "learn"
 // Expected output: 1
 const vowelTester2 = "academy"
 // Expected output: 0
 const vowelTester3 = "challenges"
 // Expected output: 2
 
+      expect(firstVowels(vowelTester1.indexOf(1))).toEqual(1)
+      expect(firstVowels(vowelTester2.indexOf(0))).toEqual(0)
+      expect(firstVowels(vowelTester3.indexOf(2))).toEqual(2)
+  } )
+ })
+//  ReferenceError: firstVowels is not defined
+
 // b) Create the function that makes the test pass.
+// psuedocode: // parameter will be an string
+// Input : we want to input strings
+//  Output: we expect to return an an index value for the associated strings.
+//  I want to find the index for "l" in learn , "a" for academy and "a" in challenges
+// I have a hunch that I can use the .indexOf method to retrive the vowels. 
+// const firstVowels = () => {
+	
+// 		// Original string
+//         const vowelTester1 = "learn";
+	
+// 		// Finding index of occurrence of 'l'
+// 		var index = vowelTester1.indexOf('l');
+		
+// 	}
+	
+
+
+
+// const firstVowels = (value, array) => {
+//     let index = -1;
+//     array.forEach((i) => {
+//       if (item === value) {
+//          index = i;
+//     }
+//   });
+//   return index;
+//  }
+
```

Comment:
I'm going to rewrite the pseudo code and the test, I think I understand this one a little better after going over it will Gene. Problem 2 is giving me the most difficulty tbh.

Reply:
<img width="1175" alt="Screenshot 2023-03-01 at 6 53 16 PM" src="https://user-images.githubusercontent.com/121983774/222319056-4fd8a0bc-1225-4122-adf1-0f600cfeff2c.png">

Reply:
<img width="1143" alt="Screenshot 2023-03-01 at 6 48 19 PM" src="https://user-images.githubusercontent.com/121983774/222319109-3ec01426-3044-40ca-b3a4-09d49bad8bac.png">